### PR TITLE
slemicro: add option to register with elemental

### PR DESCRIPTION
**Pull request by hardys**
Adds optional registration with elemental which is possible since a change landed to decouple registration from the choice of OS[1]

Note that in this case the VM should not have any k3s or RKE2 deployment enabled, since this will be handled via elemental-system-agent after the host is registered as MachineInventory with elemental

Example config:

```
ELEMENTAL_REGISTER=true
ELEMENTAL_CONFIG="$HOME/Downloads/test_registrationURL.yaml"
```

Note that `test_registrationURL.yaml` can be downloaded from the elemental UI, where a `MachineRegistration` named "test" has been created.

[1] https://github.com/rancher/elemental-operator/pull/516

### CODE REVIEW

**Review by rdoxenham — APPROVED** [review]
LGTM. Thanks!

File: slemicro/combustion/04_elemental_register.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  # Install elemental-register and elemental-system-agent
+  # Dev version currently needed for new --no-toolkit option
+  zypper ar --refresh --no-gpgcheck https://download.opensuse.org/repositories/isv:/Rancher:/Elemental:/Dev/standard/isv:Rancher:Elemental:Dev.repo
+  zypper -n in elemental-register elemental-system-agent
+  mkdir -p /etc/elemental
+  cp ./elemental_config.yaml /etc/elemental/config.yaml
+
+  # Register --no-toolkit disables OS management
+  elemental-register --config-path /etc/elemental/config.yaml --state-path /etc/elemental/state.yaml --install --no-toolkit --emulate-tpm --emulated-tpm-seed 99
+
```
Comment (hardys) [inline slemicro/combustion/04_elemental_register.sh:14]: Note that the emulated TPM is required currently - I tried to get the TPM support that landed in the latest UTM versions working, but SLEMicro won't boot - I'll try to figure that out and we can potentially resolve in a follow-up

Comment (agracey) [reply slemicro/combustion/04_elemental_register.sh:14]: I'm assuming that the TPM issue is only for UTM and not bare-metal? (I don't think I had this issue with my NUCs)

**Review by e-minguez — APPROVED** [review]
LGTM. I'll create an issue to add a couple of things:
* Check if rke2/k3s has been enabled as well and if so, fail
* Run the TPM emulation only on OSX VMs

**Pull request merged**